design/multi-homed-cluster.md

```diff
@@ -0,0 +1,67 @@
+# Rook Operators Multi-homed Cluster Network Spec
+
+## Overview
+
+This project aims to create new API to enable multiple network interface for
+Rook storage providers. Currently, Rook providers only choice is to use
+`hostNetwork` or not. The new API will be used to define networks resource for
+Rook clusters. Rook operators will be able to consume those definitions and
+manage them. Therefore, it enables more fine-grained control over storage
+providers network access.
+
+## Current Implementation
+
+To achieve non-flat networking model, Rook can choose to enable hostNetwork and
+expose host network interfaces to Storage Provider pods.
+
+EdgeFS Rook cluster network definition example:
+```yaml
+network: # cluster level networking configuration aka hostNetwork
+  serverIfName: enp2s0f0
+  brokerIfName: enp2s0f0
+```
+
+Ceph Rook cluster network definition example:
+```yaml
+network:
+  hostNetwork: true
+```
+
+Both EdgeFS and Ceph operators without specifying this configuration will always
+default to pod networking.
+
+## Proposed Design
+
+Using popular multi-plugin cni plugin such as multus, Rook operators can attach
+multiple network interfaces to pods.
+
+EdgeFS Rook Cluster definition example:
+```yaml
+network:
+  provider: multus
+  serverIfName: "flannel@net1"
+  brokerIfName: "flannel@net1"
+```
+Network provider value initially can be set to either "host" or "multus". In the
+future, works can be added to support other multi network CNI plugins such as
+knitter or genie. Omitting provider and interface names will stay defaults to
+pod networking.
+
+Interface name under "multus" provider can be written in the format of
+<network-attachment-definion-crd-name>@<interface-name>. Rook can support a
+subset of network format defined in the [Kubernetes Network Custom Resource
+Definition De-facto Standard](network-crd). Under "host" provider, anything
+but the interface name should be omitted.
+
+Ceph Rook Cluster definition example:
```

Comment:
Why would this one be different than the EdgeFS example above?

Reply:
Now that I think about it, you are right. I am not very familiar Ceph Operator, so I was trying to keep it as much close as possible to current Ceph network spec.

Reply:
I don't understand. The admin should be able to decide which network interfaces are exposed in Ceph pods. Why isn't that possible?

Reply:
I see. Sorry, I didn't quite understand your previous reply. I thought you were asking about the "host" provider limitation and provided the wrong answer.

> Why would this one be different than the EdgeFS example above?

So, this question refer to the example which I didn't list any interface, right? Currently in addition to enablling `hostNetwork`, Ceph needs to know which public and cluster network subnet it supposed to use from a ConfigMap. So, its possible, but the code aren't there yet for Ceph.

Reply:
If we agreeing, please update the design and mark it as resolved so that we can get this finalized.

Reply:
Yes, adding a quick note for Ceph should be sufficient. Thanks.